x-pack/plugins/monitoring/server/alerts/disk_usage_alert.ts

```diff
@@ -0,0 +1,361 @@
+      }),
+    },
+    ...Object.values(AlertingDefaults.ALERT_TYPE.context),
+  ];
+
+  protected async fetchData(
+    params: CommonAlertParams,
+    callCluster: any,
+    clusters: AlertCluster[],
+    uiSettings: IUiSettingsClient,
+    availableCcs: string[]
+  ): Promise<AlertData[]> {
+    let esIndexPattern = INDEX_PATTERN_ELASTICSEARCH;
+    if (availableCcs) {
+      esIndexPattern = getCcsIndexPattern(esIndexPattern, availableCcs);
+    }
+
+    const { duration, threshold } = params;
+    const stats = await fetchDiskUsageNodeStats(
+      callCluster,
+      clusters,
+      esIndexPattern,
+      duration as string,
+      this.config.ui.max_bucket_size
+    );
+
+    return stats.map((stat) => {
+      const { clusterUuid, nodeId, diskUsage, ccs } = stat;
+      return {
+        instanceKey: `${clusterUuid}:${nodeId}`,
+        shouldFire: diskUsage > threshold,
+        severity: AlertSeverity.Danger,
+        meta: stat,
+        clusterUuid,
+        ccs,
+      };
+    });
+  }
+
+  protected filterAlertInstance(alertInstance: RawAlertInstance, filters: CommonAlertFilter[]) {
+    const alertInstanceStates = alertInstance.state?.alertStates as AlertDiskUsageState[];
+    const nodeUuid = filters?.find((filter) => filter.nodeUuid);
+
+    if (!filters || !filters.length || !alertInstanceStates?.length || !nodeUuid) {
+      return true;
+    }
+
+    const nodeAlerts = alertInstanceStates.filter(({ nodeId }) => nodeId === nodeUuid);
+    return Boolean(nodeAlerts.length);
+  }
+
+  protected getDefaultAlertState(cluster: AlertCluster, item: AlertData): AlertState {
+    const currentState = super.getDefaultAlertState(cluster, item);
+    currentState.ui.severity = AlertSeverity.Warning;
+    return currentState;
+  }
+
+  protected getUiMessage(alertState: AlertState, item: AlertData): AlertMessage {
+    const stat = item.meta as AlertDiskUsageState;
+    if (!alertState.ui.isFiring) {
+      return {
+        text: i18n.translate('xpack.monitoring.alerts.diskUsage.ui.resolvedMessage', {
+          defaultMessage: `The disk usage on node {nodeName} is now under the threshold, currently reporting at {diskUsage}% as of #resolved`,
+          values: {
+            nodeName: stat.nodeName,
+            diskUsage: stat.diskUsage.toFixed(2),
+          },
+        }),
+        tokens: [
+          {
+            startToken: '#resolved',
+            type: AlertMessageTokenType.Time,
+            isAbsolute: true,
+            isRelative: false,
+            timestamp: alertState.ui.resolvedMS,
+          } as AlertMessageTimeToken,
+        ],
+      };
+    }
+    return {
+      text: i18n.translate('xpack.monitoring.alerts.diskUsage.ui.firingMessage', {
+        defaultMessage: `Node #start_link{nodeName}#end_link is reporting disk usage of {diskUsage}% at #absolute`,
+        values: {
+          nodeName: stat.nodeName,
+          diskUsage: stat.diskUsage,
+        },
+      }),
+      nextSteps: [
+        createDocLink(
```

Comment:
Love the helper!

However, I think we want to keep folks inside of Kibana as much as we can. For the CPU alert, we didn't have really any helpful charts or UIs for them to use so we had to link out to docs. This should be, IMO, a last resort.

For disk usage, I think we can link them to either Stack Monitoring pages, or other plugins to take action.

I think we specifically should do at least two things for all users:

- Link them to the Stack Monitoring ES indices listing page so they can see which indices are taking up the most disk
- Link them to the Stack Management -> ILM policy page to create a ILM policy

We can also detect if they are a cloud user, then link them to some cloud page that helps them figure out sizing.

cc @ravikesarwani to verify this or add/change anything

Reply:
So, we're going with the original list from the ticket https://github.com/elastic/kibana/issues/74819, and I also add link to SM indices